test/e2e/framework/test_context.go

```diff
@@ -303,6 +306,7 @@ func RegisterClusterFlags(flags *flag.FlagSet) {
 	flags.StringVar(&TestContext.ClusterMonitoringMode, "cluster-monitoring-mode", "standalone", "The monitoring solution that is used in the cluster.")
 	flags.BoolVar(&TestContext.EnablePrometheusMonitoring, "prometheus-monitoring", false, "Separate Prometheus monitoring deployed in cluster.")
 	flags.StringVar(&TestContext.ClusterDNSDomain, "dns-domain", "cluster.local", "The DNS Domain of the cluster.")
+	flags.StringVar(&TestContext.IPFamily, "ip-family", "ipv4", "The IP family of the cluster (ipv4, ipv6, dual-stack).")
```

Comment:
would the framework need additional logic to support the "dual-stack" case?
to my understanding this mode will be unsupported if we enable it as an option to `ip-family` here?

Reply:
why it will be unsupported?

Reply:
i should paraphrase:
"dual-stack" would be a NO-OP if we enable it here, right now, but it will work eventually?

i guess, i'm not exactly sure how it would work. if "dual-stack" is enabled would it only affect certain tests?

what would happen if the user passes "ipv6", but still attempts to run the tests that are for dual-stack?

Reply:
yeah, that´s the reason why I've suggested starting with a flag instead of an autodetecting function, this way we can have certain control. I don´t really know how the cluster would/should behave with dual-stack, but we can use the same logic that `framework.TestContext.Providers`